hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -437,9 +448,7 @@ private boolean initializeFromFilesystem(String initializationTime, List<Metadat
       HoodieData<HoodieRecord> records = fileGroupCountAndRecordsPair.getValue();
       bulkCommit(commitTimeForPartition, partitionType, records, fileGroupCount);
       metadataMetaClient.reloadActiveTimeline();
-      String partitionPath = partitionType == FUNCTIONAL_INDEX
-          ? dataWriteConfig.getFunctionalIndexConfig().getIndexName()
-          : partitionType.getPartitionPath();
+      String partitionPath = (partitionType == FUNCTIONAL_INDEX || partitionType == SECONDARY_INDEX) ? dataWriteConfig.getFunctionalIndexConfig().getIndexName() : partitionType.getPartitionPath();
```

Comment:
Maybe we can extend the partition type to hold some variable flags inside.

Reply:
Will have to supply indexing config to `MetadataPartitionType#getPartitionPath` to do so beacuse unlike functional or secondary index, all other indexes in metadata have same name as well as partition path. Functional or secondary index name are defined by user and that is why need to fetch them from functional index config.

Reply:
Yeah, probably we can add some method like `MetadataPartitionType.getPartitionPath` to simplify the code.

Reply:
Sure, will do it as part of HUDI-7691